packages/dart/pubspec.yaml

```diff
@@ -1,10 +1,10 @@
 name: parse_server_sdk
-description: The Dart SDK for Parse Platform (https://parseplatform.org)
+description: The Dart SDK for Parse Platform, based on Parse Server (https://parseplatform.org).
 version: 5.1.0
 homepage: https://github.com/parse-community/Parse-SDK-Flutter
 
 environment:
-  sdk: ">=2.18.0 <3.1.0"
+  sdk: ">=2.18.0 <4.0.0"
```

Comment:
This goes against how dart works, which is that minor versions may contain breaking changes. Setting <4.0.0 is effectively an open range, so we can as well just remove the upper bound altogether. It seems to be a conceptual bug in dart.

I think we have these options:
- a) Stay correct about the compatibility range, which is _in reality_ `<3.1.0`.
- b) Use an incorrect compatibility range (the current SDK release may not be compatible with dart 3.1.0) just to please the flawed pub.dev score points determination.
- c) Add the beta version of the next dart version to the CI and claim a potentially less incorrect range of `<=3.1.0`.

While (c) sounds a bit complicated, it may actually help us to prepare for the next minor release or dart, so we can prepare the SDK if needed before stable release.

Reply:
@mtrezza 

Why don't we proceed in the way of famous packages? See [this page](https://github.com/material-foundation/flutter-packages/commit/e7bd3c4c42df3b60b34c6c89c8de0687f535b6c1)

Reply:
That would be option (b). It would still be good if we understood why we are making that decision.

Reply:
@mtrezza What do you think we should do?

what is your opinion ?
@Nidal-Bakir

Reply:
@Nidal-Bakir 
what is your opinion?

Reply:
I think the best way to handle this is by applying the two options (b) and (c).

By testing the SDK against the beta version of Flutter/Dart and setting the upper bound to `<4.0.0`. By doing this, we will be able to detect any issues with the new beta version of Flutter/Dart immediately.

Reply:
We could try that. (b) fixes analyzer warning and (c) helps to prepare for the next release. If updating the CI job to the latest beta version for (c) turns out to be too much work we can rethink that. All on board?

Reply:
Yes!

Reply:
> We could try that. (b) fixes the analyzer warning and helps with the score and (c) helps to prepare for the next release. If updating the CI job to the latest beta version for (c) turns out to be too much work we can rethink that. All on board?

It seems good